src/reconstruction/refinement/updateMetNames.m

```diff
@@ -0,0 +1,30 @@
+function [updatedModel] = updateMetNames(originalModel, modelToUpdate)
+% Updates model.metNames in a new model, using metNames of an original
+% model.
+%
+% USAGE:
+%
+%    [updatedModel] = updateMetNames(originalModel, modelToUpdate)
+%
+% INPUTS:
+%    originalModel:      COBRA model structure with correct model.metNames
+%    modelToUpdate:      COBRA model structure that needs to have its
+%                        model.metNames updated 
+%
+% OUTPUT:
+%    updatedModel:       COBRA model structure with corrected metNames
+%
+% .. Authors:
+%       - written by Diana El Assal 27/06/2017
+
+indMets = findMetIDs(originalModel, modelToUpdate.mets);
```

Comment:
Thanks Tomas, I have now changed this function to update metNames, independently of compartment. Yes, it is intentional that every existing metName in the modelToUpdate is removed, because we have had cases when model.mets is not matching model.metNames